waves-integration/src/main/scala/com/wavesplatform/dex/domain/bytes/ByteStr.scala

```diff
@@ -39,8 +39,12 @@ final case class ByteStr(arr: Array[Byte]) {
   def takeRight(n: Long): ByteStr = drop(arr.length.toLong - n)
   def dropRight(n: Long): ByteStr = take(arr.length.toLong - n.max(0))
 
+  def canEqual(other: Any): Boolean = other.isInstanceOf[ByteStr]
+
   override def equals(a: Any): Boolean = a match {
-    case other: ByteStr => java.util.Arrays.equals(arr, other.arr)
+    case other: ByteStr =>
+      other.canEqual(this) &&
+        java.util.Arrays.equals(arr, other.arr)
```

Comment:
Also, because `ByteStr` is the final case class, we can just compare `arr` without `canEqual`